src/pharmpy/model/execution_steps.py

```diff
@@ -272,6 +264,29 @@ def _canonicalize_and_check_method(method: str) -> str:
             )
         return method
 
+    @staticmethod
+    def _canonicalize_derivatives(derivatives: tuple) -> tuple:
+        if not isinstance(derivatives, tuple):
+            raise ValueError(f"Derivatives must be given in a tuple. Got '{type(derivatives)}")
+
+        if not all(isinstance(d, tuple) for d in derivatives):
+            raise ValueError("Each derivative must be given as 'tuple'")
+
+        for d in derivatives:
+            for d_arg in d:
+                if not (isinstance(d_arg, Expr) and d_arg.is_symbol()):
+                    raise ValueError(
+                        f"Each derivative argument must be a symbol of type 'Expr'. "
+                        f"Found '{d_arg}' of type {type(d_arg)}"
+                    )
+
+        derivatives = tuple(d if isinstance(d, tuple) else (d,) for d in derivatives)
+        # Sorting of symbols creates unpredictable order
```

Comment:
Strange comment. The order becomes predictable (at least to us).

Reply:
Understandable. The comment is referring to an issue where sorting "symbols" of `Expr` type gave an unpredictable order. For this reason, strings are used instead.

Reply:
Yes, that would have been a much more accurate description